src/samples/audio/detect_test.c

```diff
@@ -653,6 +675,113 @@ static int test_keyword_prepare(struct comp_dev *dev)
 	return comp_set_state(dev, COMP_TRIGGER_PREPARE);
 }
 
+
+
+uint16_t test_keyword_get_sample_valid_bytes(struct comp_dev *dev)
+{
+	struct comp_data *cd = comp_get_drvdata(dev);
+
+	return cd->sample_valid_bytes;
+}
+
+uint32_t test_keyword_get_detected(struct comp_dev *dev)
+{
+	struct comp_data *cd = comp_get_drvdata(dev);
+
+	return cd->detected;
+}
+
+int test_keyword_set_detected(struct comp_dev *dev, uint32_t detected)
+{
+	struct comp_data *cd = comp_get_drvdata(dev);
+	int ret = 0;
+
+	cd->detected = detected;
+	return ret;
+}
+
+const int16_t *test_keyword_get_input(struct comp_dev *dev)
+{
+	struct comp_data *cd = comp_get_drvdata(dev);
+
+	/* if input is null, the function which uses
+	 * test_keyword_get_input should return
+	 * before accessing input
+	 */
+	if (!cd->input)
+		comp_err(dev, "test_keyword_get_input() null pointer");
+
+	return cd->input;
+}
+
+int16_t test_keyword_get_input_byte(struct comp_dev *dev, uint32_t index)
+{
+	struct comp_data *cd = comp_get_drvdata(dev);
+
+	if (index < 0 || index >= IN_BUFF_SIZE * sizeof(int16_t))
+		return -EINVAL;
+	return *((unsigned char *)cd->input + index);
+}
+
+int16_t test_keyword_get_input_elem(struct comp_dev *dev, uint32_t index)
+{
+	struct comp_data *cd = comp_get_drvdata(dev);
+
+	if (index < 0 || index >= IN_BUFF_SIZE)
+		return -EINVAL;
+	return cd->input[index];
+}
+
+int test_keyword_set_input_elem(struct comp_dev *dev, uint32_t index, int16_t val)
+{
+	struct comp_data *cd = comp_get_drvdata(dev);
+	int ret = 0;
+
+	if (index < 0 || index >= IN_BUFF_SIZE)
+		return -EINVAL;
+	cd->input[index] = val;
+	return ret;
+}
+
+size_t test_keyword_get_input_size(struct comp_dev *dev)
+{
+	struct comp_data *cd = comp_get_drvdata(dev);
+
+	return cd->input_size;
+}
+
+int test_keyword_set_input_size(struct comp_dev *dev, size_t input_size)
+{
+	struct comp_data *cd = comp_get_drvdata(dev);
+	int ret = 0;
+
+	cd->input_size = input_size;
+	return ret;
+}
+
+uint32_t test_keyword_get_drain_req(struct comp_dev *dev)
+{
+	struct comp_data *cd = comp_get_drvdata(dev);
+
+	return cd->drain_req;
+}
+
+int test_keyword_set_drain_req(struct comp_dev *dev, uint32_t drain_req)
+{
+	struct comp_data *cd = comp_get_drvdata(dev);
+	int ret = 0;
+
+	cd->drain_req = drain_req;
+	return ret;
+}
+
+uint32_t test_keyword_get_load_mips(struct comp_dev *dev)
+{
+	struct comp_data *cd = comp_get_drvdata(dev);
+
+	return cd->config.load_mips;
```

Comment:
Here you should check if `cd->config` is not null, before getting `load_mips`.

Reply:
cd->config is not a pointer.